kanidm_tools/src/cli/account.rs

```diff
@@ -189,37 +190,81 @@ impl AccountOpt {
-                        totp,
-                        session,
-                    ) {
-                        Ok(_) => {
-                            println!("TOTP registration success.");
-                        }
-                        Err(e) => {
-                            eprintln!("Error Completing -> {:?}", e);
+                        match client.idm_account_primary_credential_verify_totp(
+                            acsopt.aopts.account_id.as_str(),
+                            totp,
+                            session,
+                        ) {
+                            Ok(_) => {
+                                println!("TOTP registration success.");
+                                break;
+                            }
+                            Err(ClientError::TotpInvalidSha1(session)) => {
+                                eprintln!("⚠️  WARNING - It appears your authenticator app may be broken ⚠️  ");
+                                eprintln!(" The TOTP authenticator you are using is forcing the use of SHA1");
+                                eprintln!("");
+                                eprintln!(" -- If you accept this risk, and wish to proceed, type 'I am sure' ");
+                                eprintln!(" -- Otherwise press ENTER to cancel this operation");
+                                eprintln!("");
+                                eprint!("Are you sure: ");
+
+                                let mut confirm_input = String::new();
+                                if let Err(e) = io::stdin().read_line(&mut confirm_input) {
+                                    eprintln!("Failed to read from stdin -> {:?}", e);
+                                    break;
+                                };
+
+                                if confirm_input.trim() == "I am sure" {
+                                    match client.idm_account_primary_credential_accept_sha1_totp(
+                                        acsopt.aopts.account_id.as_str(),
+                                        session,
+                                    ) {
+                                        Ok(_) => {
+                                            println!("TOTP registration success.");
+                                        }
+                                        Err(e) => {
+                                            eprintln!("Error Completing -> {:?}", e);
+                                        }
+                                    };
+                                    break;
+                                } else {
+                                    eprintln!("Cancelling TOTP registration");
+                                    break;
+                                }
+                            }
+                            Err(ClientError::TotpVerifyFailed(_, _)) => {
+                                eprintln!("Incorrect TOTP code - try again");
+                                continue;
```

Comment:
As far as I have seen and tried, this will loop over and over again. Do we want this? Maybe limit the maximum tries?

Reply:
I think a limit here is a good idea. :)